frb/analysis/dmhalos_photoz.py

```diff
@@ -0,0 +1,615 @@
+        gaia_coords = SkyCoord(gaia_tab['ra'], gaia_tab['dec'], unit="deg")
+        idx, d2d, _ = gaia_coords.match_to_catalog_sky(des_coords)
+        matched_des = cat[idx][d2d<1*u.arcsec]
+        matched_gaia = gaia_tab[d2d<1*u.arcsec]
+        photom_cat = Table(np.setdiff1d(photom_cat, matched_des))
+
+    if write:
+        if outfile is None:
+            coordstr = coords.to_string(style='hmsdms', sep="", precision=2).replace(" ", "")
+            outfile = "photom_cat_J{:s}_{:0.1f}_arcmin.fits".format(coordstr,radius.to('arcmin').value) 
+        photom_cat.write(outfile, overwrite=True)
+    
+    return photom_cat
+
+def _gen_eazy_tab(photom_cat:Table, input_dir:str="eazy_in", name:str="FRB180924", out_dir="eazy_out", output_tab="no_stars_eazy.fits")->Table:
+    """
+    Run EAZY on the photometry and produce p(z) estimates.
+    Args:
+        photom_cat (Table): Photometry catalog.
+        input_dir (str, optional): Folder where EAZY config files are written.
+        name (str, optional): frb name
+        out_dir (str, optional): Folder where EAZY output is stored.
+        output_tab (str, optional): Name of the output summary table fits file.
+    Returns:
+        joined_tab (Table): EAZY results table conjoined with photom_cat.
+    """
+    
+    # Prepare EAZY
+    frb_ez.eazy_input_files(photom_cat, input_dir, name, out_dir,
+                            prior_filter="r", zmin=0.01)
+    
+    # Run it
+    logfile = os.path.join(out_dir, "eazy_run.log")
+    frb_ez.run_eazy(input_dir, name, logfile)
+
+    # read EAZY output
+    photz_file = os.path.join(out_dir, "photz.zout")
+    eazy_tab = Table.read(photz_file, format="ascii")
+    eazy_tab.rename_column('id','ID')
+    
+    # Combine the input catalog with EAZY output
+    joined_tab = join(photom_cat, eazy_tab, 'ID')
+    joined_tab.write(output_tab, overwrite=True)
+
+    return joined_tab
+
+def _create_cigale_in(photom_cat:Table, zmin:float = 0.01, zmax:float=0.35, n_z:int = 35, cigale_input:str = "cigin_minz_zfrb.fits")->Table:
+    """
+    Take the photometry table and
+    create a new table with redshifts.
+    For each galaxy, create multiple entries
+    with different redshifts from 0 to 2.
+    These redshifts will be uniformly spaced.
+    Args:
+        photom_cat (Table): Photometry catalog
+        zmin (float, optional): Minimum redshift for analysis.
+        zmax (float, optional): Maximum redshift for analysis.
+        n_z (int, optional): Number of redshift grid points.
+        cigale_input (str, optional): Nameof input file to be produced.
+    Returns:
+        stacked_photom (Table): A table with multiple groups, one for each galaxy.
+            Each entry in a group has the same photometry but different redshift values.
+            This way, CIGALE can be run on the same galaxy at multiple redshift guesses
+            in one go.
+    """
+    # Define z values
+    z_range = np.linspace(zmin, zmax, n_z) # Go up in steps of 0.01
```

Comment:
Doesn't look like steps of `0.01` to me.

Reply:
Removed this comment.